lib/cog_cmd/cfn/stack/create.rb

```diff
@@ -1,62 +1,66 @@
-    --on-failure, -f <rollback | delete | keep>
-    --timeout, -e <minutes>
-    --capabilities, -c <iam | named_iam>
-  END
-
-  def run_command
-    unless stack_name = request.args[0]
-      raise CogCmd::Cfn::ArgumentError, "You must specify a stack name AND a template name."
+      # options
+      @stack_params = request.options['param']
+      @tags = request.options['tag']
+      @policy = request.options['policy']
+      @notify = request.options['notify']
+      @on_failure = request.options['on-failure']
+      @timeout = request.options['timeout']
+      @capabilities = request.options['capabilities']
     end
 
-    unless template_name = request.args[1]
-      raise CogCmd::Cfn::ArgumentError, "You must specify a stack name AND a template name."
+    def run_command
+      raise(CogCmd::Cfn::ArgumentError, "You must specify a stack name AND a template name.") unless stack_name
+      raise(CogCmd::Cfn::ArgumentError, "You must specify a stack name AND a template name.") unless template_name
+
+      response.template = 'stack_show'
+      response.content = create_stack
     end
 
-    client = Aws::CloudFormation::Client.new()
-
-    cf_params = Hash[
-      [
-        [ :stack_name, stack_name ],
-        [ :template_url, template_url(template_name) ],
-        param_or_nil([ :parameters, process_parameters(request.options["param"]) ]),
-        param_or_nil([ :tags, process_tags(request.options["tag"]) ]),
-        param_or_nil([ :stack_policy_url, policy_url(request.options["policy"]) ]),
-        param_or_nil([ :notification_arns, request.options["notify"] ]),
-        param_or_nil([ :on_failure, process_on_failure(request.options["on-failure"]) ]),
-        param_or_nil([ :timeout_in_minutes, request.options["timeout"] ]),
-        param_or_nil([ :capabilities, process_capabilities(request.options["capabilities"]) ])
-      ].compact
-    ]
-
-    client.create_stack(cf_params)
-    client.describe_stacks(stack_name: stack_name).stacks[0].to_h
-  end
+    private
 
-  private
+    def create_stack
+      client = Aws::CloudFormation::Client.new()
 
-  def process_parameters(params)
-    params.map do |p|
-      param = p.strip.split('=')
-      { parameter_key: param[0],
-        parameter_value: param[1] }
+      params = {
+        stack_name: stack_name,
+        template_url: template_url(template_name),
+        parameters: process_parameters(stack_params),
+        tags: process_tags(tags),
+        stack_policy_url: policy_url(policy),
+        notification_arns: notify,
+        on_failure: process_on_failure(on_failure),
+        timeout_in_minutes: timeout,
+        capabilities: process_capabilities(capabilities)
+      }.reject { |_key, value| value.nil? }
+
+      client.create_stack(params)
+      client.describe_stacks(stack_name: stack_name).stacks[0].to_h
     end
-  end
 
+    def process_parameters(params)
+      return nil unless params
```

Comment:
Super minor but the ruby idiom for this is to omit `nil` like `return unless params`